source/fe/mapping_q.cc

```diff
@@ -396,21 +386,24 @@ MappingQ<dim,spacedim>::fill_fe_subface_values (const typename Triangulation<dim
        :
        &data);
 
-  const unsigned int n_q_points=q.size();
-  this->compute_fill_face (cell, face_no, sub_no,
+  const unsigned int n_q_points = quadrature.size();
+  this->compute_fill_face (cell, face_no, subface_no,
                            n_q_points,
                            QProjector<dim>::DataSetDescriptor::
-                           subface (face_no, sub_no,
+                           subface (face_no, subface_no,
                                     cell->face_orientation(face_no),
                                     cell->face_flip(face_no),
                                     cell->face_rotation(face_no),
                                     n_q_points,
                                     cell->subface_case(face_no)),
-                           q.get_weights(),
+                           quadrature.get_weights(),
                            *p_data,
-                           quadrature_points, JxW_values,
-                           exterior_forms, normal_vectors, jacobians,
-                           inverse_jacobians);
+                           output_data.quadrature_points,
+                           output_data.JxW_values,
+                           output_data.boundary_forms,
+                           output_data.normal_vectors,
+                           output_data.jacobians,
+                           output_data.inverse_jacobians);
```

Comment:
Wouldn't it make sense to pass `FEValuesData<dim,spacedim>` on to `compute_fill_face`? It is not the public interface but it would facilitate things if one wanted to compute additional quantities on faces (e.g. the Jacobian gradients). (Or do you intend to do this in the second part?)